internal/pproflogging/pproflogging_test.go

```diff
@@ -124,3 +219,140 @@ func TestDebug_newProfileConfigs(t *testing.T) {
+			expectConfigurationCount: 3,
+			profileKey:               "block",
+			profileFlagKey:           "rate",
+			expectProfileFlagExists:  true,
+			expectProfileFlagValue:   "10",
+			expectError:              nil,
+		},
+		{
+			inArgs:                   "block=rate=10:cpu:mutex=10",
+			expectConfigurationCount: 3,
+			profileKey:               "cpu",
+			profileFlagKey:           "rate",
+			expectProfileFlagExists:  false,
+		},
+		{
+			inArgs:                   "block=rate=10:cpu:mutex=10",
+			expectConfigurationCount: 3,
+			profileKey:               "mutex",
+			profileFlagKey:           "10",
+			expectProfileFlagExists:  true,
+		},
+		{
+			inArgs:                       "mutex=10",
+			expectConfigurationCount:     1,
+			profileKey:                   "cpu",
+			profileFlagKey:               "10",
+			expectProfileConfigNotExists: true,
+		},
+	}
+
+	for i, tc := range tcs {
+		t.Run(fmt.Sprintf("%d: %q", i, tc.inArgs), func(t *testing.T) {
+			pmp, err := LoadProfileConfig(ctx, tc.inArgs)
+			require.ErrorIs(t, tc.expectError, err)
+			if err != nil {
+				return
+			}
+			val, ok := pmp[tc.profileKey]
+			require.Equalf(t, tc.expectProfileConfigNotExists, !ok, "expecting key %q to %t exist", tc.profileKey, !tc.expectProfileConfigNotExists)
+			if tc.expectProfileConfigNotExists {
+				return
+			}
+			flagValue, ok := val.GetValue(tc.profileFlagKey)
+			require.Equal(t, tc.expectProfileFlagExists, ok, "expecting key %q to %t exist", tc.profileKey, tc.expectProfileFlagExists)
+			if tc.expectProfileFlagExists {
+				return
+			}
+			require.Equal(t, tc.expectProfileFlagValue, flagValue)
+		})
+	}
+}
+
+//nolint:gocritic
+func saveLockEnv(t *testing.T) {
+	t.Helper()
+
+	oldEnv := os.Getenv(EnvVarKopiaDebugPprof)
+
+	t.Cleanup(func() {
+		// restore the old environment
+		t.Setenv(EnvVarKopiaDebugPprof, oldEnv)
+	})
+}
+
+func TestErrorWriter(t *testing.T) {
+	eww := &ErrorWriter{mx: 5, err: io.EOF}
+	n, err := eww.WriteString("Hello World")
+	require.ErrorIs(t, io.EOF, err)
+	require.Equal(t, 5, n)
+	require.Equal(t, "Hello", string(eww.bs))
+}
+
+// ErrorWriter allows injection of errors into the write stream.  There are a few
+// failures in PPROF dumps that are worth modeling for tests ([io.EOF] is one)
+// For use specify the error, ErrorWriter.err, and byte index, ErrorWriter.mx,
+// in which it should occur.
+type ErrorWriter struct {
+	bs  []byte
+	mx  int
+	err error
+}
+
+func (p *ErrorWriter) Write(bs []byte) (int, error) {
+	n := len(bs)
+
+	if len(bs)+len(p.bs) > p.mx {
+		// error will be produced at p.mx
+		// so don't return any more than
+		// n
+		n = p.mx - len(p.bs)
+	}
+
+	// append the bytes to the local buffer just
+	// in case someone wants to know.
+	p.bs = append(p.bs, bs[:n]...)
+	if n < len(bs) {
+		// here we assume that any less than len(bs)
+		// bytes written returns an error.  This
+		// allows setting ErrorWriter up once
+		// to produce an error after multiple
+		// writes
```

Comment:
It seems that there's a corner case here though when `len(bs) == 0`, is that the case?
That is, if an error was set and previously returned, a subsequent call to `p.Write([]byte{})` would return without an error? Is that intended behavior?